### PR TITLE
Fix coreNode error when fetching public data with no user session 

### LINKED ISSUE #802: Fix coreNode error when fetching public data with no user session
As first encountered with https://github.com/markmhx/humans/issues/12, I'm getting `TypeError: Cannot read property 'coreNode' of undefined` upon calling `getFile` and reaching: https://github.com/blockstack/blockstack.js/blob/950dc4fc838099835ce0592f245453c770663b84/src/storage/index.ts#L253

`sessionData.userData` is undefined and throws an error instead of getting to use `caller.appConfig.coreNode`

cc @zone117x since he appears to be the last person to touch this part of the code

### PR DESCRIPTION
**Pull request by ahsan-javaid**
## Description
This PR fixes the issue of coreNode error when fetching public data with no user session in storage package.

This PR closes the issue #802

## Type of Change
- [ ] New feature
- [X] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
oN

## Testing information
- Added test case to verify the coreNode utilized from app config, not from user session as userData is undefined
- `npm run test`

## Checklist
- [ ] Code is commented where needed
- [X] Unit test coverage for new or modified code paths
- [X] `npm run test` passes
- [ ] Changelog is updated
- [X] Tag 1 of @yknl or @zone117x for review

### CODE REVIEW

File: packages/storage/tests/storage.test.ts [hunk 1]
```diff
@@ -257,6 +257,25 @@ test('getFile unencrypted, unsigned', async () => {
...
+    app: 'http://localhost:8080',
+    decrypt: false,
+  };
+  userSession.appConfig.coreNode = appSpecifiedCoreNode;
+  let storage = new Storage({ userSession });
+  try {
+    //Get file will utilize cordNode url from app config not from userSession
+    await storage.getFile(path, options);
+  } catch {}
+  expect(true).toEqual(true);
```
Comment (zone117x) [inline packages/storage/tests/storage.test.ts]: Does this really test the behavior? Seems like this test would pass regardless of the above fix

**Comment by markmhendrickson** [comment]
This PR is great to see! 🙏

File: packages/storage/tests/storage.test.ts (hunk 1 above)
Comment (ahsan-javaid) [reply packages/storage/tests/storage.test.ts]: Test case is updated to test the behaviour.

**Review by zone117x — APPROVED** [review]
Thanks @ahsan-javaid!

**Review by reedrosenbluth — APPROVED** [review]
Thanks!

**Pull request merged**